src/Symfony/Component/Messenger/Transport/Sender/SendersLocator.php

```diff
@@ -23,17 +27,29 @@
  */
 class SendersLocator implements SendersLocatorInterface
 {
-    private $senders;
+    private $sendersMap;
+    private $sendersLocator;
+    private $useLegacyLookup = false;
     private $sendAndHandle;
 
     /**
-     * @param SenderInterface[][] $senders
-     * @param bool[]              $sendAndHandle
+     * @param string[][] $sendersMap An array, keyed by "type", set to an array of sender aliases
+     * @param ContainerInterface Locator of senders, keyed by sender alias
+     * @param bool[]     $sendAndHandle
      */
-    public function __construct(array $senders, array $sendAndHandle = [])
+    public function __construct(array $sendersMap, /*ContainerInterface*/ $sendersLocator = null, array $sendAndHandle = [])
     {
-        $this->senders = $senders;
-        $this->sendAndHandle = $sendAndHandle;
+        $this->sendersMap = $sendersMap;
+
+        if (is_array($sendersLocator) || null === $sendersLocator) {
+            @trigger_error(sprintf('The 2nd argument to "%s::__construct()" requires ContainerInterface 2nd argument. Not passing that was deprecated in Symfony 4.3 and will be required in Symfony 5.0.', __CLASS__), E_USER_DEPRECATED);
+            $this->sendersLocator = new ServiceLocator([]);
+            $this->sendAndHandle = $sendersLocator;
+            $this->useLegacyLookup = true;
```

Comment:
BC added so that FW bundle 4.2 can work with Messenger 4.3 (otherwise cross-version tests fail)